src/Simulation/Core/TypeExtensions.cs

```diff
@@ -144,6 +147,55 @@ public static Type[] GetTupleFieldTypes(this Type arg)
             }
         }
 
+        /// <summary>
+        /// Given an <see cref="object"/>, retrieve its non-qubit fields as a string.
+        /// Returns null if no non-qubit fields found.
+        /// </summary>
+        public static string? GetNonQubitArguments(this object o)
```

Comment:
Renamed this from `ArgsToString` to more correctly describe what this method does.